src/codegen/llvm/llvm_benchmark.cpp

```diff
@@ -0,0 +1,157 @@
+    // First, set the output stream for the logs.
+    set_log_output();
+
+    // Then, record the time taken for building the LLVM IR module.
+    codegen::CodegenLLVMVisitor visitor(mod_filename,
+                                        output_dir,
+                                        llvm_build_info.opt_passes,
+                                        llvm_build_info.use_single_precision,
+                                        llvm_build_info.vector_width);
+    generate_llvm(visitor, node);
+
+    // Finally, run the benchmark and log the measurements.
+    run_benchmark(visitor, node);
+}
+
+void LLVMBenchmark::generate_llvm(codegen::CodegenLLVMVisitor& visitor,
+                                  const std::shared_ptr<ast::Program>& node) {
+    // First, visit the AST to build the LLVM IR module and wrap the kernel function calls.
+    auto start = std::chrono::high_resolution_clock::now();
+    visitor.visit_program(*node);
+    visitor.wrap_kernel_functions();
+    auto end = std::chrono::high_resolution_clock::now();
+
+    // Log the time taken to visit the AST and build LLVM IR.
+    std::chrono::duration<double> diff = end - start;
+    *log_stream << "Created LLVM IR module from NMODL AST in " << std::setprecision(PRECISION)
+                << diff.count() << "\n\n";
+}
+
+std::vector<std::string> LLVMBenchmark::get_cpu_features() {
+    std::string cpu(llvm::sys::getHostCPUName());
+
+    llvm::SubtargetFeatures features;
+    llvm::StringMap<bool> host_features;
+    if (llvm::sys::getHostCPUFeatures(host_features)) {
+        for (auto& f: host_features)
+            features.AddFeature(f.first(), f.second);
+    }
+    return features.getFeatures();
+}
+
+void LLVMBenchmark::run_benchmark(codegen::CodegenLLVMVisitor& visitor,
+                                  const std::shared_ptr<ast::Program>& node) {
+    // Set the codegen data helper and find the kernels.
+    auto codegen_data = codegen::CodegenDataHelper(node, visitor.get_instance_struct_ptr());
+    std::vector<std::string> kernel_names;
+    visitor.find_kernel_names(kernel_names);
+
+    // Get feature's string and turn them off depending on the backend.
+    std::vector<std::string> features = get_cpu_features();
+    *log_stream << "Backend: " << backend << "\n";
+    if (backend == "avx2") {
+        // Disable SSE.
+        *log_stream << "Disabling features:\n";
+        disable("sse", features);
+        disable("sse2", features);
+        disable("sse3", features);
+        disable("sse4.1", features);
+        disable("sse4.2", features);
+    } else if (backend == "sse2") {
+        // Disable AVX.
+        *log_stream << "Disabling features:\n";
+        disable("avx", features);
+        disable("avx2", features);
+    }
+
+    std::string features_str = llvm::join(features.begin(), features.end(), ",");
+    std::unique_ptr<llvm::Module> m = visitor.get_module();
+    runner::Runner runner(std::move(m), features_str);
```

Comment:
I'm not sure if it would make sense to also measure the `JIT` compilation time as well

Reply:
@iomaganaris This is something to look at for sure. I think that it is not significant anyway (we just compile < 1000 instructions), but may still be worth considering. (I will add it/look at it when adding more metrics, such as min/max times, etc.)

Reply:
Yes, that's what I had in mind. In case we had to measure the JIT compilation time only measuring the construction of `runner::Runner` (and more explicitly the `JITDriver::init()`) would be enough, right?

Reply:
I plan to add my "favourite table" and thats where we could log this.